sdk/src/Core/Amazon.Runtime/Internal/Transform/JsonUnmarshallerContext.cs

```diff
@@ -409,14 +418,19 @@ private void UpdateContext()
 
         #endregion
 
-        public JsonData ToJsonData()
+        public JsonDocument ToJsonDocument(ref StreamingUtf8JsonReader streamingReader)
         {
-            var data = JsonMapper.ToObject(jsonReader);
+            JsonDocument document = null;
+            streamingReader.PassReaderByRef((ref Utf8JsonReader reader) =>
+            {
+                document = JsonDocument.ParseValue(ref reader);
```

Comment:
That is a great point. Then we would have to actually read all the data from the stream and fill the buffer and then pass in the reader, instead.

Reply:
So originally this method was called `ToJsonData`
https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/Amazon.Runtime/Internal/Transform/JsonUnmarshallerContext.cs#L412

and since the reader was part of the context, it just used the context's reader. But I don't see it being called anywhere either actually 

https://github.com/search?q=repo%3Aaws%2Faws-sdk-net+ToJsonData&type=code

So I'm okay with just removing this method and then also the PassReaderByRef, method since this was the only reason I created this method in the first place